Sources/TuistDependencies/SwiftPackageManager/Utils/SwiftPackageManagerGraphGenerator.swift

```diff
@@ -404,8 +404,8 @@ extension ProjectDescription.Settings {
         settings: [PackageInfo.Target.TargetBuildSettingDescription.Setting]
     ) throws -> Self? {
         var headerSearchPaths: [String] = []
-        var defines: [String: String] = [:]
-        var swiftDefines: [String] = []
+        var defines: [String: String] = ["SWIFT_PACKAGE": "1"]
```

Comment:
It'd be great to add a test here that ensures that we are setting that build setting. The test will prevent regressions in the future.

Reply:
It is tested by the test that uses the `Sources/TuistDependenciesTesting/DependenciesGraph/DependenciesGraph+TestData.swift` I have changed in this PR

Reply:
Ouch, I see. In that case, I suggest taking a different approach. The fact that you are implicitly describing the test through test data is not obvious for a contributor. I think tests should be very explicit:

```swift
func loaded_targets_have_the_swift_package_compiler_macro() {
  // Test implementation
}
```

As a contributor to the project, if I see that test is failing, I know what might have caused it and can fix it quickly. If I change something in the test data and tests start failing, it'd be hard for me to know what's causing the tests to fail.